File: src/TKDHubAPI.Application/DTOs/Promotion/CreatePromotionDto.cs
namespace TKDHubAPI.Application.DTOs.Promotion;
/// <summary>
/// DTO for creating a new promotion.
/// </summary>
public class CreatePromotionDto
{
    [Required]
    public int StudentId { get; set; }

    public int RankId { get; set; }

    public DateTime PromotionDate { get; set; }

    [Required]
    public int CoachId { get; set; }

    [MaxLength(1000)]
    public string? Notes { get; set; }

    [Required]
    public int DojaangId { get; set; }
}

File: Frontend/tkd_hub_web/src/app/types/CreatePromotionDto.ts
export type CreatePromotionDto = {
	studentId: number;
	rankId: number;
	promotionDate: string; // ISO date string, e.g. "2025-06-15T21:53:06.830Z"
	coachId: number;
	notes?: string;
	dojaangId: number;
};
